rust/src/resource_owner_password_grant_authenticator.rs

```diff
@@ -0,0 +1,77 @@
+use crate::error::{Error, VResult};
+use crate::{authenticator::Authenticator, message::OpenIdConnectTokenResponse};
+use async_trait::async_trait;
+use reqwest::{StatusCode, Url};
+
+pub(crate) struct ResourceOwnerPasswordGrantAuthenticator {
+    client_id: String,
+    user_name: String,
+    password: String,
+    token_url: Url,
+}
+
+impl ResourceOwnerPasswordGrantAuthenticator {
+    pub fn new(client_id: &str, user_name: &str, password: &str, token_url: &Url) -> Self {
```

Comment:
token_url should have a default value that the customer does not need to set. I suggest to use the builder pattern here, to allow it to be an option.

If you need to own all the values anyway, why expect a reference?

Reply:
TODO:
* New auth module
* Move all implemtations of Authentication to auth
* Implement Authentication for client_credentials_grant
* Add auth to builder
* Remove parameters from connect
* Methods: Pass values instead of references if ownership is required

Reply:
* Use builder for Authenticator options (https://crates.io/crates/derive_builder), for easy to use values